### PR TITLE
[bugfix] Fix index mismatch adding events when global df with multiple IDs is used

### LINKED ISSUE #879: [bug] index mismatch with events in global setup
By @rbeldagarcia in #827
 
Hello! Thanks for the update, some features are very useful like to allow create a global model with a unique data frame.

I write this comment to report some observation of the new version.  I have tried to migrate my models written in 0.3.2 to 0.4.1 and I have found some issues.
In particular, these models are Global model with events, and the problems appear in: 
Could you confirm if the requirements.txt is updated? I would like to validate that my environment is correct before reporting this as a bug. In particular, I have tried with:

pandas=1.3.5
neuralprophet=0.4.1
numpy=1.21.6

I have tried to create a new environment with requirements.txt but it reports a error trying to install torch>=1.8.0

df_utils.py (line 1006)    

```
    for event in config_events.keys():
        event_feature = pd.Series([0.0] * df.shape[0])
        event_feature.index = df.index  #added to fix the errors, without this it reports NaN when the events are added to the dataframe in global model. 
        				 #The problem is that if df.index != event_feature.index then it return NaN when try to do: df[event] = event_feature


        # events_df may be None in case ID from original df is not provided in events df
        if events_df is None:
            dates = None
        else:
            dates = events_df[events_df.event == event].ds
            event_feature.iloc[df.ds.isin(dates)] = 1.0 #Added 'iloc' to support events that are only present in some 'ID' groups
        df[event] = event_feature
    return df
```
    
I attach you the code and a screen shoot 

[Neuralprophet041.zip](https://github.com/ourownstory/neural_prophet/files/9818669/Neuralprophet041.zip)

_Originally posted by @rbeldagarcia in https://github.com/ourownstory/neural_prophet/discussions/827#discussioncomment-3913276_

### PR DESCRIPTION
**Pull request by judussoari**
Fixes #879

### CODE REVIEW

File: neuralprophet/df_utils.py [hunk 1]
```diff
@@ -1027,6 +1027,7 @@ def convert_events_to_features(df, config_events, events_df):
             dates = None
         else:
             dates = events_df[events_df.event == event].ds
+            df.reset_index(drop=True, inplace=True)
```
[the remark this answers is not in the document]
Comment (ourownstory) [reply neuralprophet/df_utils.py:1030]: never mind, lets keep this

**Review by ourownstory — CHANGES_REQUESTED** [review]
Good!
Let's also do an index reset in _make_future_dataframe for both events and future regressors.
That will also resolve #799

**Comment by judussoari** [comment]
> 

index reset is already performed in `_make_future_dataframe()` for both `events_df` and `regressors_df`. Ready to merge!

**Review by ourownstory — APPROVED** [review]
LGTM!

**Pull request merged**